File: CS/WindowsFormsApplication1/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;
using DevExpress.DataAccess.Sql;

namespace WindowsFormsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            XtraReport1 report = new XtraReport1();
            report.Parameters.Add(new DevExpress.XtraReports.Parameters.Parameter() { Name = "CatID", Type = typeof(Int32), Value = 2, Description = "CategoryID" });

            SqlDataSource ds = report.DataSource as SqlDataSource;
            TableQuery tq = ds.Queries[0] as TableQuery;
            

            DevExpress.DataAccess.Sql.QueryParameter queryParameter1 = new DevExpress.DataAccess.Sql.QueryParameter();
            queryParameter1.Name = "p1";
            queryParameter1.Type = typeof(DevExpress.DataAccess.Expression);
            queryParameter1.Value = new DevExpress.DataAccess.Expression("[Parameters.CatID]");
            tq.Parameters.Add(queryParameter1);
            tq.FilterString = "CategoryID = ?p1";

            ReportPrintTool printTool = new ReportPrintTool(report);
            printTool.ShowPreviewDialog();
        }

    }

}

File: VB/WindowsFormsApplication1/Form1.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI
Imports DevExpress.DataAccess.Sql

Namespace WindowsFormsApplication1
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            Dim report As New XtraReport1()
            report.Parameters.Add(New DevExpress.XtraReports.Parameters.Parameter() With {.Name = "CatID", .Type = GetType(Int32), .Value = 2, .Description = "CategoryID"})

            Dim ds As SqlDataSource = TryCast(report.DataSource, SqlDataSource)
            Dim tq As TableQuery = TryCast(ds.Queries(0), TableQuery)


            Dim queryParameter1 As New DevExpress.DataAccess.Sql.QueryParameter()
            queryParameter1.Name = "p1"
            queryParameter1.Type = GetType(DevExpress.DataAccess.Expression)
            queryParameter1.Value = New DevExpress.DataAccess.Expression("[Parameters.CatID]")
            tq.Parameters.Add(queryParameter1)
            tq.FilterString = "CategoryID = ?p1"

            Dim printTool As New ReportPrintTool(report)
            printTool.ShowPreviewDialog()
        End Sub

    End Class

End Namespace
